src/array_api_stubs/_draft/array_object.py

```diff
@@ -63,9 +69,10 @@ def ndim(self: array) -> int:
         out: int
             number of array dimensions (axes).
         """
+        ...
 
     @property
-    def shape(self: array) -> Tuple[Optional[int], ...]:
+    def shape(self) -> tuple[int | None, ...]:
```

Comment:
Just to say more for myself than anyone else—I initially wasn't too sure if we wanted to use the py3.9+ type hint features, but:
* Forgot that `from future import __annotations__` makes this work for py3.8
* py3.8 doesn't get updates end of next year
* The scientific ecosystem already is adopting py3.9+